src/client/reseed.cc

```diff
@@ -220,222 +225,5 @@ bool Reseed::FetchStream(
-    const std::string alpha = "abcdefghijklmnopqrstuvwxyz";
-    std::regex regex("([-"+alpha+"0-9+._']{1,254})@((?:[-"+alpha+"0-9]+.)+["+alpha+"|(i2p)]{2,})");
-    if (!std::regex_search(m_Data->signer_id.data(), regex)) {
-      LOG(error) << "SU3: invalid signer ID";
-      return false;
-    }
-    // Save position
-    m_Data->signature_position = m_Stream.Tellg();
-    // Prepare to read in both content length + signature length
-    m_Data->content_length += m_Data->signature_position;
-    m_Data->content.resize(m_Data->content_length);
-    m_Data->signature.resize(m_Data->signature_length);
-    // Read in content and signature for verification against signer ID
-    m_Stream.Seekg(0, std::ios::beg);
-    m_Stream.Read(*m_Data->content.data(), m_Data->content.size());
-    m_Stream.Read(*m_Data->signature.data(), m_Data->signature.size());
-    // Go back to prepare for RI extraction
-    m_Stream.Seekg(m_Data->signature_position, std::ios::beg);
-    // Our content position is the same as signature position
-    m_Data->content_position = m_Data->signature_position;
-  } catch (const std::exception& e) {
-    LOG(error) << "SU3: caught exception '" << e.what() << "' during preparation";
-    return false;
-  }
-  LOG(debug) << "SU3: preparation successful";
-  return true;
-}
-
-bool SU3::VerifySignature() {
-  // Get signing keys from extracted/processed SU3 certs
-  auto signing_key_it = m_SigningKeys.find(m_Data->signer_id.data());
-  if (signing_key_it == m_SigningKeys.end()) {
-    LOG(error)
-      << "SU3: certificate for " << m_Data->signer_id.data() << " not loaded";
-    return false;
-  }
-  // Verify hash of content data and signature
-  switch (m_Data->signature_type) {
-    case kovri::core::SIGNING_KEY_TYPE_RSA_SHA512_4096: {
-      kovri::core::RSASHA5124096RawVerifier verifier(signing_key_it->second);
-      verifier.Update(m_Data->content.data(), m_Data->content.size());
-      if (!verifier.Verify(m_Data->signature.data())) {
-        LOG(error) << "SU3: signature failed";
-        return false;
-      }
-      break;
-    }
-    // TODO(unassigned): see #160
-    // Note: this is currently redundant since signature type was
-    // tested during stream preparation. We'll leave this here
-    // because it will eventually be useful.
-    default:
-      LOG(error)
-        << "SU3: signature type " << m_Data->signature_type << " is not supported";
-      return false;
-  }
-  LOG(debug) << "SU3: verification successful";
-  return true;
-}
-
-bool SU3::ExtractContent() {
-  LOG(debug) << "SU3: unzipping stream";
-  kovri::client::ZIP zip(
-      m_Stream.Str(),
-      m_Data->content_length,
-      m_Data->content_position);
-  if (!zip.Unzip()) {
-    LOG(error) << "SU3: unzip failed";
-    return false;
-  }
-  // Get unzipped RI's for Reseed
-  m_RouterInfos = zip.m_Contents;
-  LOG(debug) << "SU3: extraction successful";
-  return true;
-}
-
```

Comment:
I don't agree with moving this all out of the implementation file. Why did you want to do this?